flang/test/Lower/OpenMP/empty-omp-parallel.f90

```diff
@@ -25,9 +27,28 @@ program parallel
 !LLVMIRDialect-NEXT: omp.terminator
 !LLVMIRDialect-NEXT: }
 
-!LLVMIR: call void{{.*}}@__kmpc_fork_call{{.*}}@[[OMP_OUTLINED_FN:.*]] to {{.*}}
-!LLVMIR: define internal void @[[OMP_OUTLINED_FN]]
+!LLVMIR: call i32 @__kmpc_global_thread_num(%struct.ident_t* @{{.*}})
+!$OMP PARALLEL NUM_THREADS(16)
+!$OMP END PARALLEL
+        num_threads = 4
+!$OMP PARALLEL NUM_THREADS(num_threads)
 !$OMP END PARALLEL
+
+!FIRDialect: omp.parallel num_threads(%{{.*}} : i32) {
+!FIRDialect-NEXT: omp.terminator
+!FIRDialect-NEXT: }
+
+!LLVMIRDialect: omp.parallel num_threads(%{{.*}} : !llvm.i32) {
+!LLVMIRDialect-NEXT: omp.terminator
+!LLVMIRDialect-NEXT: }
+
+!LLVMIR: %[[GLOBAL_THREAD_NUM:.*]] = call i32 @__kmpc_global_thread_num(%struct.ident_t* @{{.*}})
+!LLVMIR-DAG:  call void @__kmpc_push_num_threads(%struct.ident_t* @{{.*}}, i32 %[[GLOBAL_THREAD_NUM]], i32 16)
+!LLVMIR-DAG: %[[GLOBAL_THREAD_NUM1:.*]] = call i32 @__kmpc_global_thread_num(%struct.ident_t* @{{.*}})
+!LLVMIR-DAG:  call void @__kmpc_push_num_threads(%struct.ident_t* @{{.*}}, i32 %[[GLOBAL_THREAD_NUM1]], i32 %{{.*}})
+
+!LLVMIR-DAG: call void{{.*}}@__kmpc_fork_call{{.*}}@[[OMP_OUTLINED_FN:.*]] to {{.*}}
+!LLVMIR-DAG: define internal void @[[OMP_OUTLINED_FN]]
```

Comment:
Are checks missing for one of the omp.parallel's with num_threads?

Reply:
It was there, but now I've re-organized it better for readability/uniformity